dandi/tests/test_models.py

```diff
@@ -154,3 +166,265 @@ def test_autogenerated_titles():
     assert schema["title"] == "Asset Meta"
     assert schema["properties"]["schemaVersion"]["title"] == "Schema Version"
     assert schema["definitions"]["PropertyValue"]["title"] == "Property Value"
+
+
+def datacite_post(datacite, doi):
+    """
+    posting the datacite object and returning the status codes
+    of requests.post and requests.get
+    """
+
+    # removing doi in case it exists
+    _clean_doi(doi)
+
+    # checking f I'm able to create doi
+    rp = requests.post(
+        "https://api.test.datacite.org/dois",
+        json=datacite,
+        headers={"Content-Type": "application/vnd.api+json"},
+        auth=("DARTLIB.DANDI", os.environ["DATACITE_DEV_PASSWORD"]),
+    )
+
+    # checking if i'm able to get the url
+    rg = requests.get(url=f"https://api.test.datacite.org/dois/{doi}/activities")
+
+    # cleaning url
+    _clean_doi(doi)
+    return rp.status_code, rg.status_code
+
+
+def _clean_doi(doi):
+    """removing doi, ignoring the status code"""
+    requests.delete(
```

Comment:
You should call `.raise_for_status()` to ensure that the request is successful.

Reply:
this is sometimes run "just in case" before creating a new doi, so it can return an error

Reply:
shouldn't it matter what kind of error it is?  e.g. if we disconnected an not removed DOI -- we should try again.  `grep` for `tenacity` to see how we orchestrate retries ATM.

Note: we now have 57 DOIs in test. instance of datacite, some of them seems to be for the same dataset(s).  I wonder if some of those are due to not carrying if cleanup succeeded or not, and shouldn't we actually cleanup most if not all of them? ;-)